### PR TITLE
Add pipeline tuner component and dependencies.

### PR DESCRIPTION
**Pull request by ruizhuanguw**
Add an example using flaml to tune an AML pipeline.

### CODE REVIEW

File: test/pipeline_tuning_example/tuner/env.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+channels:
+- defaults
+dependencies:
+- python=3.8
+- pip:
+  - azure-ml-component[notebooks]==0.9.10.post1
+  - azureml-dataset-runtime==1.39.0
+  - hydra-core==1.1.1
+  - flaml[blendsearch,ray]==1.0.7
```
Comment (sonichi) [inline test/pipeline_tuning_example/tuner/env.yaml:9]:
```suggestion
  - flaml[blendsearch,ray]==1.0.9
```

File: test/pipeline_tuning_example/tuner/tuner_func.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        else:
+            run_metrics = list(metrics.values())
+
+            new_metric = run_metrics[0]['eval_binary_error']
+
+            if type(new_metric) == list:
+                new_metric = new_metric[-1]
+
+            print(f'eval_binary_error: {new_metric}')
+            if 'old_metric' not in locals() or new_metric != old_metric:
```
Comment (sonichi) [inline test/pipeline_tuning_example/tuner/tuner_func.py:51]: define `old_metric` first.

**Review by sonichi — COMMENTED** [review]
Please fix the bug and test the code. Self-check the following:

- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

File: test/pipeline_tuning_example/tuner/tuner_func.py (hunk 2 above)
Comment (ruizhuanguw) [reply test/pipeline_tuning_example/tuner/tuner_func.py]: Update the logic to avoid the issue.

File: test/pipeline_tuning_example/tuner/tuner_func.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    while not stop:
+        # get status
+        status = run._core_run.get_status()
+        print(f'status: {status}')
+
+        # get metrics
+        metrics = run._core_run.get_metrics(recursive=True)
+        if metrics == {}:
+            pass
+        else:
```
Comment (sonichi) [inline test/pipeline_tuning_example/tuner/tuner_func.py:31]:
```suggestion
        if metrics:
```

File: website/docs/Examples/Tune-AzureML pipeline.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    metric = best_trial.metric_analysis[hp_metric][mode]
+    print(f"n_trials={len(analysis.trials)}")
+    print(f"time={time.time()-start_time}")
+    print(f"Best {hp_metric}: {metric:.4f}")
+    print(f"Best coonfiguration: {best_trial.config}")
+```
+
+### Interact with AzureML pipeline jobs
+
+The interaction between FLAML and AzureML pipeline jobs lie in `tuner_func.run_with_config`.
```
Comment (sonichi) [inline website/docs/Examples/Tune-AzureML pipeline.md:153]:
```suggestion
The interaction between FLAML and AzureML pipeline jobs is in `tuner_func.run_with_config`.
```

File: website/docs/Examples/Tune-AzureML pipeline.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
+# Tune - AzureML pipeline
```
Comment (sonichi) [inline website/docs/Examples/Tune-AzureML pipeline.md:1]: Please rename this file into "Tune-AzureML-pipeline.md"

File: test/pipeline_tuning_example/tuner/tuner_func.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    search_space = {
+        "train_config.n_estimators": flaml.tune.randint(50, 200),
+        "train_config.learning_rate": flaml.tune.uniform(0.01, 0.5),
+    }
+
+    hp_metric = "eval_binary_error"
+    mode = "max"
+    num_samples = 2
+
+
```
Comment (sonichi) [inline test/pipeline_tuning_example/tuner/tuner_func.py:67]: remove

File: website/docs/Examples/Tune-AzureML-pipeline.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+We are now ready to set up the HPO job for the AzureML pipeline, including:
+
+- config the HPO job,
+- set up the interaction between the HPO job and the training job.
+
+These two steps are done in `tuner/tuner_func.py`.
+
+### Set up the tune job
+
+`tuner_func.tune_pipeline` sets up the search space, metric to optimize, mode, et al.
```
Comment (sonichi) [inline website/docs/Examples/Tune-AzureML-pipeline.md:111]:
```suggestion
`tuner_func.tune_pipeline` sets up the search space, metric to optimize, mode, etc.
```

**Review by sonichi — APPROVED** [review]
looks good except minor things.

File: test/pipeline_tuning_example/tuner/tuner_func.py (hunk 6 above)
Comment (ruizhuanguw) [reply test/pipeline_tuning_example/tuner/tuner_func.py:67]: @sonichi we will need mode and num_samples to specify the HPO config. I put all the HPO config set-up together from lline 58 to line 65. Could you clarify which line we should remove?

Comment (sonichi) [reply test/pipeline_tuning_example/tuner/tuner_func.py:67]: sorry, I meant removing line 67 only.

**Review by tgautam23 — APPROVED** [review]
